### ISSUE TITLE
Can not start on arm platform

### ISSUE THREAD

**Issue by easyyu** [hivemq/hivemq-community-edition#239]
I'm trying with Orange pi 2E with Armbian latest image. Version of Java is 11.
After successful compile:
```
root@orangepiplus2e:/opt# git clone https://github.com/hivemq/hivemq-community-edition.git
Cloning into 'hivemq-community-edition'...
remote: Enumerating objects: 227, done.
remote: Counting objects: 100% (227/227), done.
remote: Compressing objects: 100% (144/144), done.
remote: Total 13204 (delta 71), reused 157 (delta 31), pack-reused 12977
Receiving objects: 100% (13204/13204), 8.79 MiB | 4.44 MiB/s, done.
Resolving deltas: 100% (8589/8589), done.
root@orangepiplus2e:/opt# cd hivemq-community-edition
root@orangepiplus2e:/opt/hivemq-community-edition# 
root@orangepiplus2e:/opt/hivemq-community-edition# ./gradlew clean packaging
Downloading https://services.gradle.org/distributions/gradle-6.3-bin.zip
.........10%..........20%..........30%.........40%..........50%..........60%.........70%..........80%..........90%..........100%

Welcome to Gradle 6.3!

Here are the highlights of this release:
 - Java 14 support
 - Improved error messages for unexpected failures

For more details see https://docs.gradle.org/6.3/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

######################################################################################################
You can not use the latest changes of or modify the hivemq-extension-sdk.
Please checkout the hivemq-extension-sdk repository next to the hivemq-community-edition repository.
Execute the following command from your project directory:
git clone https://github.com/hivemq/hivemq-extension-sdk.git ../hivemq-extension-sdk
You can also clone your fork:
git clone https://github.com/<replace-with-your-fork>/hivemq-extension-sdk.git ../hivemq-extension-sdk
######################################################################################################

> Task :clean UP-TO-DATE
> Task :cleanPackaging
> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :processResources
> Task :classes
> Task :shadowJar
> Task :copyJar
> Task :copyPackaging
> Task :copyXml
> Task :packaging

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 11m 32s
9 actionable tasks: 8 executed, 1 up-to-date
```

when I start, I get error:

```
root@orangepiplus2e:/opt/hivemq-community-edition/build/package/hivemq-ce-2020.6# ./bin/run.sh
-------------------------------------------------------------------------

                  _    _  _              __  __   ____
                 | |  | |(_)            |  \/  | / __ \ 
                 | |__| | _ __   __ ___ | \  / || |  | |
                 |  __  || |\ \ / // _ \| |\/| || |  | |
                 | |  | || | \ V /|  __/| |  | || |__| |
                 |_|  |_||_|  \_/  \___||_|  |_| \___\_\

-------------------------------------------------------------------------

  HiveMQ Start Script for Linux/Unix v1.13

-------------------------------------------------------------------------

  HIVEMQ_HOME: /opt/hivemq-community-edition/build/package/hivemq-ce-2020.6

  JAVA_OPTS:  -Djava.net.preferIPv4Stack=true --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.nio=ALL-UNNAMED --add-opens java.base/sun.nio.ch=ALL-UNNAMED --add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED --add-exports java.base/jdk.internal.misc=ALL-UNNAMED -Djava.security.egd=file:/dev/./urandom -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=9010 -Dcom.sun.management.jmxremote.local.only=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -XX:+CrashOnOutOfMemoryError -XX:+HeapDumpOnOutOfMemoryError

  JAVA_VERSION: 11

-------------------------------------------------------------------------

2021-02-01 03:24:52,993 INFO  - Starting HiveMQ Community Edition Server
2021-02-01 03:24:53,022 INFO  - HiveMQ version: 2020.6
2021-02-01 03:24:53,022 INFO  - HiveMQ home directory: /opt/hivemq-community-edition/build/package/hivemq-ce-2020.6
2021-02-01 03:24:53,311 INFO  - Log Configuration was overridden by /opt/hivemq-community-edition/build/package/hivemq-ce-2020.6/conf/logback.xml
2021-02-01 03:24:55,241 INFO  - This HiveMQ ID is Nrna1
2021-02-01 03:24:58,268 ERROR - Uncaught Error:
java.lang.UnsatisfiedLinkError: /tmp/librocksdbjni10049088761556698760.so: /tmp/librocksdbjni10049088761556698760.so: cannot open shared object file: No such file or directory (Possible cause: can't load IA 32-bit .so on a ARM-bit platform)
        at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
        at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
        at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
        at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
        at java.base/java.lang.Runtime.load0(Runtime.java:768)
        at java.base/java.lang.System.load(System.java:1837)
        at org.rocksdb.NativeLibraryLoader.loadLibraryFromJar(NativeLibraryLoader.java:79)
        at org.rocksdb.NativeLibraryLoader.loadLibrary(NativeLibraryLoader.java:57)
        at org.rocksdb.RocksDB.loadLibrary(RocksDB.java:67)
        at org.rocksdb.RocksDB.<clinit>(RocksDB.java:38)
        at org.rocksdb.FlushOptions.<clinit>(FlushOptions.java:14)
        at com.hivemq.persistence.payload.PublishPayloadRocksDBLocalPersistence.<clinit>(PublishPayloadRocksDBLocalPersistence.java:48)
        at com.hivemq.persistence.payload.PublishPayloadRocksDBLocalPersistence$$FastClassByGuice$$bfd4a565.newInstance(<generated>)
        at com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:89)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
        at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:42)
        at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:65)
        at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:113)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:213)
        at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:184)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:111)
        at com.google.inject.Guice.createInjector(Guice.java:87)
        at com.hivemq.bootstrap.ioc.GuiceBootstrap.persistenceInjector(GuiceBootstrap.java:142)
        at com.hivemq.HiveMQServer.main(HiveMQServer.java:139)
```

Whats wrong?

**Comment by SgtSilvio** [comment]
Hi @easyyu 
Unfortunately the FILE_NATIVE persistence is currently not supported on ARM.
To avoid this, please use the FILE persistence:
* Change FILE_NATIVE to FILE in https://github.com/hivemq/hivemq-community-edition/blob/master/src/main/java/com/hivemq/configuration/service/InternalConfigurations.java#L126
* Change FILE_NATIVE to FILE https://github.com/hivemq/hivemq-community-edition/blob/master/src/main/java/com/hivemq/configuration/service/InternalConfigurations.java#L157
* Then recompile `./gradlew packaging`

Instead you can also use in-memory persistence.

I hope this helps. We will keep this issue open until you report back.

**Comment by tehbilly** [comment]
I'm facing this same issue, even when setting:

```xml
<persistence>
    <mode>in-memory</mode>
</persistence>
```

**Comment by DC2-DanielKrueger** [comment]
Can u try:
`<persistence>
    <mode>file</mode>
</persistence>`
please?

In-memory mode should work actually. But i think the problem could be the static field inside PublishPayloadRocksDBLocalPersistence (FLUSH_OPTIONS). A quick fix would be to make it not static but a field and initiate it in the constructor instead. The reason is that the static field gets initiated as soon as the class is loaded - even when it is never injected really by Guice. I hope that helps till we fix it (if my thesis is right)

Best regards,
Daniel from HiveMQ

**Comment by MicWalter** [comment]
Should be resolved with #252
